examples/service_discovery.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import aiohttp
+import asyncio
+
+import asab
+import asab.web
+import asab.api
+import asab.zookeeper
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+asab.Config.add_defaults(
+	{
+		"web": {
+			"listen": "0.0.0.0 8089",
+		},
+
+		"zookeeper": {
+			"path": "example",
+			"servers": "localhost:2181"
+		},
+	}
+)
+
+
+class MyApplication(asab.Application):
+
+	async def initialize(self):
+		# Initialize web server
+		self.add_module(asab.web.Module)
+		websvc = self.get_service("asab.WebService")
+		self.WebContainer = asab.web.WebContainer(websvc, "web")
+
+		# Initialize zookeeper
+		self.add_module(asab.zookeeper.Module)
+		self.ZooKeeperService = self.get_service("asab.ZooKeeperService")
+		self.ZooKeeperContainer = asab.zookeeper.ZooKeeperContainer(self.ZooKeeperService, "zookeeper")
+
+		# Service Discovery is part of ApiService and needs its full functionality
+		self.ASABApiService = asab.api.ApiService(self)
+		self.ASABApiService.initialize_web(self.WebContainer)
+		self.ASABApiService.initialize_zookeeper(self.ZooKeeperContainer)
+
+		# Localize Service Discovery Service
+		self.ServiceDiscoveryService = self.get_service("asab.ServiceDiscoveryService")
+
+		self.WebContainer.WebApp.router.add_get('/locate', self.locate_self)
+
+	async def locate_self(self, request):
+		await asyncio.sleep(2)
+		urls = await self.ServiceDiscoveryService.locate(appclass="MyApplication")
```

Comment:
The app is locating itself, which is not really lovely example... But I don't find running a bunch of microservices to fill in the zookeeper as a simple example

Reply:
If there is a proper in-code documentation, it is ok.